packages/ember/tests/routing/router_service_test/currenturl_lifecycle_test.js

```diff
@@ -0,0 +1,209 @@
+          this._super();
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/').then(() => {
+        assert.equal(routerService.get('currentURL'), '/');
+      });
+    }
+
+    ['@test RouterService#currentURL is correctly set for child route'](assert) {
+      assert.expect(1);
+
+      let routerService;
+
+      this.registerRoute('parent.child', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/child').then(() => {
+        assert.equal(routerService.get('currentURL'), '/child');
+      });
+    }
+
+    ['@test RouterService#currentURL is correctly set after transition'](assert) {
+      assert.expect(1);
+
+      let routerService;
+
+      this.registerRoute('parent.child', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        },
+
+        afterModel() {
+          this.transitionTo('parent.sister');
+        }
+      }));
+
+      return this.visit('/child').then(() => {
+        assert.equal(routerService.get('currentURL'), '/sister');
+      });
+    }
+
+    ['@test RouterService#currentURL is correctly set on each transition'](assert) {
+      assert.expect(3);
+
+      let routerService;
+
+      this.registerRoute('parent.child', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/child')
+        .then(() => {
+          assert.equal(routerService.get('currentURL'), '/child');
+
+          return this.visit('/sister');
+        })
+        .then(() => {
+          assert.equal(routerService.get('currentURL'), '/sister');
+
+          return this.visit('/brother');
+        })
+        .then(() => {
+            assert.equal(routerService.get('currentURL'), '/brother');
+        });
+    }
+
+    ['@test RouterService#currentURL is not set during lifecycle hooks'](assert) {
+      assert.expect(2);
+
+      return this.visit('/')
+        .then(() => {
+          assert.deepEqual(results, [
+            [null, 'beforeModel', null],
+            [null, 'model', null],
+            [null, 'afterModel', null]
+          ]);
+
+          results = [];
+
+          return this.visit('/child');
+        })
+        .then(() => {
+          assert.deepEqual(results, [
+            ['parent.index', 'beforeModel', '/'],
+            ['parent.index', 'model', '/'],
+            ['parent.index', 'afterModel', '/']
+          ]);
+        });
+    }
+
+    ['@test RouterService#currentURL is correctly set with component'](assert) {
+      assert.expect(1);
+
+      return this.visit('/').then(() => {
+        this.assertText('/');
+      });
```

Comment:
Can you add an update and reset phase to this (kinda like the INUR stuff for the glimmer tests)?

I'm thinking something like:

```js
.then(() => routerService.transitionTo('child'))
.then(() => this.assertText('/child'))
.then(() => routerService.transitionTo('/'))
.then(() => this.assertText('/'))
```